### PR TITLE
Server generated ids

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR moves ID generation to the server side, and modifies the insert flow to return and take into account the new ID on the frontend side of the application.

#### Card link
[Focalboard card](https://community-daily.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/47aa9bb4-6967-4a96-83c7-11bd6b20f1eb/ca1a5441-4c1c-4d0e-856e-88cc744576c8?c=230e9332-ad73-4f95-85d4-3b26654efcf0)

### CODE REVIEW

File: server/utils/utils.go [hunk 1]
```diff
@@ -54,3 +55,59 @@ func StructToMap(v interface{}) (m map[string]interface{}) {
...
+			return existingID
+		}
+		return id
+	}
+
+	getExistingOrNewID := func(id string) string {
+		if existingID, ok := newIDs[id]; ok {
+			return existingID
+		}
+		return NewID(IDTypeBlock)
```
Comment (wiggin77) [inline server/utils/utils.go:100]: I created a PR with a method to convert block types to IDType.  https://github.com/mattermost/focalboard/pull/1715   That can be used to pass the correct IDType here.

Comment (wiggin77) [reply server/utils/utils.go:100]: That PR was revised here https://github.com/mattermost/focalboard/pull/1744 to move the new method to a better place.

**Review by jespino — APPROVED** [review]
LGTM

File: server/utils/utils.go (hunk 1 above)
Comment (mgdelacroix) [reply server/utils/utils.go]: Updated on 71cc8f7, PTAL

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:  Great work @mgdelacroix

**Pull request merged**